### PR TITLE
Initial translation updates

### PR DESCRIPTION
**Pull request by Avvilimento**
Call me crazy but I kinda want to know the story to this game, such as it is. The provided machine translations are... well, they're bad. Not "it put the words in the wrong order" or "it messed up the pronouns" bad, but "I really don't care about getting a return on my investment" and "No I don't care about your website" bad. If you know you know, and if you don't then you clearly weren't reading. Seen as I was going to be doing this for myself, and no-one else has beaten me to it, I decided I would upload my current efforts to to community.

Plus it was a good opportunity to learn Python. Combining the 10,000+ files into one to find duplicate lines (I expected some, but only ~160,000 or so of the 450,000+ lines are unique), propagating translations across those duplicates, and being able to correctly split it back up again - all at the press of a button - makes the process almost seem bearable. Still going to be a long, frustrating journey to the end, though.

I should mention that I don't speak Japanese, I cannot read Japanese, and I don't have an in-depth understanding of their culture. I know a few Japanese words (a few more since starting this) and have a broad overview of their day-to-day but that's about all. As such I've based my work on a combination of machine translation (Google, Bing, and DeepL, which I believe was used for the original machine translation), individual word search, and a bit of guess-work. Due to the presence of onomatopoeia and the more colloquial nature of some language used the translation quality varies a bit, with chapter 9 of Ai's story (c-10/00/8**) feeling particularly poor, but I still feel it to be reasonably accurate.

With that said, this is a translation - not an interpretation. I've made no attempt to "localize" the story in any way. So some people may find it a little "sweet" as a result of my maintaining the general politeness of Japanese dialogue ("Miss Minase", "Miss Otonari", etc), a little dry due to my lack of "flavour" describing the more erotic portions (seriously, they use the term "Ochinchin", which is considered a childish way of referring to the penis; "Asoko", which seems to basically mean "down there"; "Omanko", which is an extremely vulgar way to refer to the vagina; and the rest of the terms are almost entirely clinical - save one or two mentions of "meat stick"), and perhaps even a little strange as I've tried to maintain regional nuances (such as the use of "sadist" and "masochist" for S and M respectively, despite their far stronger English connotations). So if someone wants to re-interpret my work - spice it up a bit - be my guest.

Oh, and I use EnGB, so expect to see extra Us and Ss.

### CODE REVIEW

**Comment by ManlyMarco** [comment]
Hello, and thanks for the PR! 

The quality of translations is definitely better overall. The current main story translations were submitted by many different people with different proficiency levels, so they can definitely use some cleanup. The "stand-alone" characters are slowly being translated by an experienced translator (e.g. #48 #47) and don't need much attention outside of the PRs.

Main issue I can see is the quoting style (we have agreed to not put text inside quotes). e.g. `﻿「きみはなかなか怪我しないわね。やっぱり身体が頑丈なのかしら？」="You don't get hurt very often, do you? I wonder if your body is strong enough?"` should be without the outer `"`.

Also be careful with blanket copying translations - depending on the context and personality the translation might end up being a lot different between identical JP strings. It could also result in merge conflicts for the big PRs, so I'll wait for #48 to get merged before merging this one.

There's a full DeepL translation available [here](https://www.patreon.com/posts/better-english-45341283) but it has minimal cleaning and is meant to be used as a fallback.

**Comment by ShadowTsuki** [comment]
I'll try to skim over this when I have the time but looks promising enough. 
One thing I did catch was that there shouldn't be any CHOICE after the "=". 
### CHOICE:部活の成果を確認してもらえますか？=CHOICE:Can you check the results of my club activities?
Should be: ### CHOICE:部活の成果を確認してもらえますか？=Can you check the results of my club activities?
I suggest joining the discord, it's easier to talk if you have any questions about structure or the translation itself.

**Comment by Avvilimento** [comment]
It's probably superfluous to join the Discord. I'm doing this mostly as a time filler and curiosity, and honestly expect I'll get distracted by something else before I'm finished.

I can very quickly make the requested changes to the submission (another benefit of the bulk file method). Like I said, I was just translating as directly as I could manage from the source. To that end I should also ask about lines surrounded by parenthesis and the lines from an "absent" character. Ordinarily I'd do the latter in italics but I'm not sure that's an option in this situation, so for now I've just used square brackets.

**Comment by ShadowTsuki** [comment]
Just put it out there if you had any further questions. 

The lines in brackets for thoughts? Use regular brackets ().
“Absent” character as in the lines without any sort of brackets? The same, nothing around them. 

We’re just following the format to Party for consistency’s sake (although I do question why they decided to translate the way they did). I guess the coloured lines were enough indication for whether someone was speaking vs the narration.

**Comment by Avvilimento** [comment]
By "absent" characters I mean the ones they've used hollow brackets for in the original. At least as far as Ai's story is concerned it's pretty much the writing in the notebook (chapter 1), the incoming text messages (chapter 4), and the "far" side of the phone call (chapter 8).  eg: 『作戦って？』=[What was the plan?]

I didn't actually realise that Party was in English. It's probably been considered and dismissed already, but just in case... is it possible to dump the text from Party into the same basic format as this? Assuming no major changes to sequence were made it would be possible to make a script to automatically "translate" to it in a way compatible with the translation plugin, at least for the purposes of a fallback translation when nothing better is available.

**Comment by ShadowTsuki** [comment]
Ah, we use " " for those ones. 

Using Party's official translation we determined to be pirating, so we don't put it here. The rip is out there, pretty sure the majority who have the original version has it, it's just not available here.

**Comment by Avvilimento** [comment]
Weird, though I responded...

Fair enough on the Party dialogue.

Made what I suspect to be the final change to the formatting and fixed another issue I found on the way. Should be ready for vetting now. The more I think about it the more I feel like I should have taken some liberties with chapter 9's terminology, but like I said I just translated what was there initially. I guess that comes back to the context and personality bit which ManlyMarco mentioned.

**Comment by ShadowTsuki** [comment]
I'll look through it and make any changes probably around next weekend. 
One thing I wasn't sure to mention before, but I think we should keep honorifics/suffixes (chan, san, sensei, etc) as they are, as well as more widely known weeb terms. No need to change anything for this PR, I'll do it, but something to note for future translations.

**Comment by ShadowTsuki** [comment]
Ok, one thing I noticed that NEEDS to be fixed is that the Japanese text gets edited, specifically the ellipses. 
image
I'll look over the rest, but I won't commit any edits until afterwards just to make sure.

**Comment by Avvilimento** [comment]
Ah damn. I thought I'd been clever enough to avoid doing that. Just giving a quick look over it, the only affected ones are the single sets - doubles, triples, and so on don't look to be wrong. But that's still a decent number of lines in a decent number of files.

If you give me a few minutes I'll take care of those. I'll have to check my working master as well.

**Comment by Avvilimento** [comment]
OK, I think I got them all. If I'm right they were entirely limited to Lina's dialogue, and all of them were double ellipses that had been converted. That's at least somewhat comforting, but I'm still going to check my master file to make sure.

File: translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt
Comment (ManlyMarco) [inline translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt:3]: Is removing the names in some files a problem?

Comment (Avvilimento) [reply translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt:3]: That particular reference is for the title on her speech window. Without having that in each file you could wind up having only the time the reference is there using her translated name and the rest would fall back to the original, though it's possible if you encountered her more than one session it may keep translating depending upon how it caches (I would suspect it is per-file, not per-session).

The automatic translation plugin would mask this, however, as it would add it to the global automatic translation file. If you're using it, anyway.

The only way you could remove those duplicate references without issue (and without the auto-translator) would be if the translation plugin supported "bulk" files. By that I mean instead of having 31 subdirectories under /adv/scenario/c-1 with 29 files in them, you have just a single Translation.txt in /adv/scenario/c-1 which contains all of the translations required for Lina. But I don't know that it would work, since I don't know that the plugin supports such a thing.

Theoretically you could already make a complete "master translation" file by replacing the _Preprocessors.txt or _Substitutions.txt file with one that contains the complete game text. This would remove all duplication, shrinking down to only the unique lines, but would hinge fairly heavily on the plugin's ability to parse 160K+ lines.

Comment (GeBo1) [reply translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt:3]: Removing these causes more problems than it fixes, and under the covers doesn't work at all the same way as what happens when they aren't there. I'm strongly opposed to removing these. These files are instantaneous, have much better specificity, and are applied before the game engine ever sees the text.  The fallback to the auto-translate doesn't apply until the text is sent to the screen. 

Since `TranslationSync` works on these files so keeping these entries in sync is no problem.

I've floated your suggestion about having fallback/master files per-personality, etc. multiple times (I had even coded it up and tested it), but there didn't seem to be any interest at the time so I dropped it.

**Review by GeBo1 — CHANGES_REQUESTED** [review]
Please restore character names to the files under RedirectedResources (seem my comment elsewhere)

File: translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt
Comment (Avvilimento) [reply translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt:3]: Oh, wait, I see what's going on here. I think. Just let me ramble a bit to see if I've got it...

I only put those name references into files which had it when I dumped the text initially. My understanding being that they weren't referenced for other files (the game likely automatically inserts an automatic [H名]), thus aren't required. The files that have it correspond to scenes which take place in a "conversation setting", for lack of a better term - where the view gets letterboxed and the girl the player is talking to is brought into centre focus - as the name in use can change there, such as when the Swimming Club Captain or Ai's mother shows up.

The trouble comes in that the translator plugin receives that implied "[H名]" and still has to look that data up. Because there's no global cache, or if there is there's still the chance of hitting that reference before said cache has been seeded, the reference will get directed to either the auto-translator, the original Japanese text, just pass through directly as "[H名]" because it has no translation, or performs some kind of exception.

So, to prevent untoward behaviour, any file which is used to draw text onto a "speech window" which includes a character name, at least for the purposes of story characters, needs to have a provided translation for said name.

**Comment by Avvilimento** [comment]
Each of Lina's files should have a name translation again. Also found some more ellipses that somehow got changed in the original Japanese text within the story listing and adjusted the formatting of the rest to match the changes already made by @ShadowTsuki (basically just removed the trailing period, where present).  

Fairly sure I didn't undo any of the already done changes, but do verify that.

File: translation/en/RedirectedResources/assets/abdata/adv/scenario/c-10/00/0/translation.txt
Comment (GeBo1) [inline translation/en/RedirectedResources/assets/abdata/adv/scenario/c-10/00/0/translation.txt:125]: These are still missing (also if you could put them back at the end of the file it's easier to track the history)

Comment (Avvilimento) [reply translation/en/RedirectedResources/assets/abdata/adv/scenario/c-10/00/0/translation.txt:125]: The ones in Ai's story (adv/scenario/c-10) were never gone. They're just where the dumped data put them, which I assume is immediately before they're used.

Easy enough to move them all to the end for consistency, though.

File: translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt
Comment (GeBo1) [reply translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt:3]: Sort of. When the game loads the asset from the unity3d file containing those strings TextResourceRedirector replaces the matching strings from the specific matching asset file before the game engine ever sees it. A one time replacement when the asset is loaded.   The game engines substitution engine (what replaces the inline variables like `[H名]` will only ever see the translated text if it's already handled and the translation engine doesn't need to do anything at runtime then.

When the strings aren't replaced at the asset level, the auto-translation will pick it up once it's going to become visible to the user (assigned to a UI element). Usually this is fast enough you don't notice, but it does it each and every time. Also in the case the names are used in substitution, the text from the asset is inserted BEFORE the auto-translation has a chance to see it, so you can get the original Japanese inserted into otherwise translated strings causing the whole string to go to the auto-translator, which may change/reorder the parts that are already translated.

Thanks for putting them back!

**Review by GeBo1 — APPROVED** [review]

**Comment by Avvilimento** [comment]
I don't know that @ShadowTsuki had finished going over it. But I guess they can always do another pull request with any updates.

Time for the next ones. I'll make them separate this time to ease the review process (hindsight I shouldn't have done the story indexes and two different characters at the same time).

**Comment by GeBo1** [comment]
Ah, sorry @ShadowTsuki, I'd missed your comment about coming back to this.  @Avvilimento feel free to drop the review into draft if you know you're waiting on something or know you're going to make more changes.

**Comment by Avvilimento** [comment]
Gonna show how new I am to Git here, but... how would I go about doing that? I don't have any changes that I'm personally looking to do, but if it helps others make tweaks for things I've missed (or just things they would have done differently) it's possibly a good thing to know how to do it.

**Comment by GeBo1** [comment]
Once you file the request there's a button for in on GitHub.  I can't remember the exact wording

**Comment by Avvilimento** [comment]
Gotcha. I should be able to figure it out if it's needed.

**Comment by ShadowTsuki** [comment]
One thing I noticed is that the "Restored name reference" update changed the ellipses in the original Japanese text again to "..." and wasn't fixed in the later updates.
I'm currently going over all the files so I'll fix it as I go through, just wanted to make sure @Avvilimento was aware of it for the future.
Screen Shot 2021-04-27 at 2 21 45 PM

**Comment by Avvilimento** [comment]
Damnit... Somewhere in between my master file reverted. I guess the how and why isn't important (probably something to do with the mess of editors I have open) but it still bothers me... Sorry about that.

I'll double check what I'm working on at the moment (I think I'm done with Reina, mostly done with Nonoka) to make sure I haven't done it again, and I'll correct my masters... again...

**Pull request merged**
